### ISSUE TITLE
unnecessary-constructor parameters for super call

### ISSUE THREAD

**Issue by Bublade** [palantir/tslint#4594]
### Rule Suggestion
So there is already an unnecessary-constructor but it does not check for this construction:
```typescript
constructor(requiredParam: string) {
    super(requiredParam);
}
```
While this is also unnecessary and it should be great if this would also trigger the unnecessary-constructor rule.

I wasn't sure in which category I should have posted this. This is isnt realy a "new rule" but wasn't a "bug" either.

**Comment by JoshuaKGoldberg** [comment]
Agreed! This could become a breaking change, as the rule explicitly states it only checks for blank constructors. Let's make this an opt-in flag for now, and plan to enable it by default in a later version.

**Comment by rafaelss95** [comment]
What should happen in the following case?

```ts
class X {
  constructor(private param1: number, param2: number) {}
}

export class Y extends Z {
  constructor(param1: number) {
    super(param1, 2);
  }
}
```

**Comment by JoshuaKGoldberg** [comment]
@rafaelss95 did you mean that `Y extends X`, not `Z`?

If so: `Y`'s constructor seems unnecessary per this request. `Y` wouldn't have access to `this.param1` no matter what, and the constructor doesn't use either parameter. 

If not: `Y`'s constructor would still seem unnecessary, since it doesn't use either parameter.

In both cases, `X`'s constructor is not unnecessary, as `param1` is stored as `this.param1`.

**Comment by rafaelss95** [comment]
@JoshuaKGoldberg I've misspelled `Z` (should be` X`).

Let me see if I got the request here: check for `constructors` that have **only** parameters (not parameter properties) and/or its body is composed only by a super call, right?

<hr>

If so, when the proposed option is enabled, the following cases should fail:

https://github.com/palantir/tslint/blob/613c311a58b95097b786d3838673da0022f41734/test/rules/unnecessary-constructor/test.ts.lint#L46

https://github.com/palantir/tslint/blob/613c311a58b95097b786d3838673da0022f41734/test/rules/unnecessary-constructor/test.ts.lint#L65

**Comment by JoshuaKGoldberg** [comment]
> ```ts
>  constructor() {
>      super();
>  }
> ```

Yup, this should fail. So should an equivalent `constructor(a) { super(a); }`, as removing the constructor wouldn't change any logic.

> `constructor(public x: number) { } `

That constructor is not unnecessary and should not cause a complaint from the rule. It contains a `this.x = x;` so it does actual work. That code is the same as:

```
public x: number;

constructor(x: number) {
    this.x = number;
}
```
